### ISSUE TITLE
Fuzzing-found errors

### ISSUE THREAD

**Issue by thenhnn** [phip1611/tar-no-std#18]
I found some new errors with AFL.

These are files that cause tar_no_std to panic: 
[crashes-new.tar.gz](https://github.com/phip1611/tar-no-std/files/15203545/crashes-new.tar.gz)

```
thread 'main' panicked at /home/user/src/tar-no-std/src/archive.rs:306:44:
attempt to subtract with overflow
```

It's still in progress, I will update if it will find something new.

**Comment by phip1611** [comment]
> attempt to subtract with overflow

Fixed in v0.3.1.

However, I think I spend now enough time in making this crate more bullet-proof. Thanks, your input helped a lot! :)

We should not forget that in use-cases this library is made for, one doesn't need to expect malicious or intentionally malformed Tar archives. Although correct, these scenarios are rather obscure.


If you don't find any more critical bugs, which I'm happy to fix, feel free to open PRs and fix them yourself.

**Comment by thenhnn** [comment]
> > attempt to subtract with overflow
> 
> Fixed in v0.3.1.
> 
> However, I think I spend now enough time in making this crate more bullet-proof. Thanks, your input helped a lot! :)
> 
> We should not forget that in use-cases this library is made for, one doesn't need to expect malicious or intentionally malformed Tar archives. Although your intentions are correct, these scenarios are rather obscure.
> 
> If you don't find any more critical bugs, which I'm happy to fix, feel free to open PRs and fix them yourself.

I haven't found anything in 50minutes of fuzzing, so I don't think there will be something more.

Thank you, too!

However, I personally think that don't panic policy is important, especially for no_std libraries :)
